modyn/supervisor/internal/triggers/datadrifttrigger.py

```diff
@@ -200,16 +238,32 @@ def _run_detection(self) -> tuple[bool, dict[str, MetricResult]]:
 
         # Compute embeddings
         assert self.embedding_encoder is not None
+
+        # tbd. reuse the embeddings as long as the reference window is not updated
         reference_embeddings = get_embeddings(self.embedding_encoder, reference_dataloader)
         current_embeddings = get_embeddings(self.embedding_encoder, current_dataloader)
         reference_embeddings_df = convert_tensor_to_df(reference_embeddings, "col_")
         current_embeddings_df = convert_tensor_to_df(current_embeddings, "col_")
 
         drift_results = {
-            **self.evidently_detector.detect_drift(reference_embeddings_df, current_embeddings_df),
-            **self.alibi_detector.detect_drift(reference_embeddings, current_embeddings),
+            **self.evidently_detector.detect_drift(reference_embeddings_df, current_embeddings_df, is_warmup),
+            **self.alibi_detector.detect_drift(reference_embeddings, current_embeddings, is_warmup),
         }
+
+        # make the final decisions with the decision engines
+        for metric_name, metric_result in drift_results.items():
+            distance = (
+                metric_result.distance
+                if isinstance(metric_result.distance, float)
+                else float(mean(metric_result.distance))
```

Comment:
why is this aggregation needed? maybe we can add a comment here

Reply:
We could implement sth. like with an EnsemblePolicy (I think we already have the configuration models for that). However, for drift, we don't want to recompute embeddings if we want to evaluate two different metrics (e.g., mmd and classifier); that's why we do aggregation here.

Reply:
I still don't fully understand this. We already have the ensemble logic in case we have e.g. a threshold decision engine with MMD and a threshold decision policy with euclidean, and then we can e.g. do majority vote. I don't understand here yet why we averaging multiple metrics is required and what is the use case.

Reply:
metrics can also live on completely different scales I suppose (like 5000 and 0.005 - it really depends on the metric itself). Can this just be averaged?

Reply:
I think i got confused myself when writing the comment there. This averaging is only creating scalar distance values from an array of distance values returned by an alibi-detect metric (some of the metrics there don't return a single distance value but a distance value for every sample)

Reply:
Ahhh, ok. Maybe we should move this logic then to the detector level? Because from the detector interface, the returned metric should be a scalar value, not a list, shouldn't it?

Reply:
Yes, let's do that, I also thought of that, but back then, I wanted the full detector results in the logs. Now I'm confident of the results so we can do that.